opencensus/exporters/stats/stackdriver/internal/stackdriver_utils.cc

```diff
@@ -181,14 +181,20 @@ void SetMetricDescriptor(
 std::vector<google::monitoring::v3::TimeSeries> MakeTimeSeries(
     absl::string_view metric_name_prefix,
     const google::api::MonitoredResource& monitored_resource,
+    const std::unordered_map<std::string, google::api::MonitoredResource>&
+        per_view_monitored_resource,
     const opencensus::stats::ViewDescriptor& view_descriptor,
     const opencensus::stats::ViewData& data,
     absl::string_view opencensus_task) {
   // Set values that are common across all the rows.
   auto base_time_series = google::monitoring::v3::TimeSeries();
   base_time_series.mutable_metric()->set_type(
       MakeType(metric_name_prefix, view_descriptor.name()));
-  if (monitored_resource.type().empty()) {
+  if (per_view_monitored_resource.find(view_descriptor.name()) !=
```

Comment:
This does the search twice. How about something like:
```
auto iter = find(...);
if (iter != end) { set(*iter); } else ...
```

Reply:
Ah, yes! Thanks for catching this. Done.